src/routes/UserSettings/CommunitySettingsTab/CommunitySettingsTab.js

```diff
@@ -1,51 +1,206 @@
+      {showAddAffiliations ? <AddAffiliation close={this.toggleAddAffiliations} save={this.saveAffiliation} /> : (
+        <div styleName='add-affiliation' onClick={this.toggleAddAffiliations}>
+          <div styleName='plus'>+</div>
+          <div>Add new affiliation</div>
+        </div>
+      )}
     </div>
+    )
   }
-}
 
-export function CommunityControl ({ membership, leaveCommunity }) {
-  const leave = () => {
-    if (window.confirm(`Are you sure you want to leave ${community.name}?`)) {
-      leaveCommunity(community.id)
-    }
+  deleteAffiliation = (affiliationId) => {
+    const { deleteAffiliation } = this.props
+    const { affiliations } = this.state
+
+    deleteAffiliation(affiliationId)
+      .then(res => {
+        let errorMessage, successMessage
+        if (res.error) errorMessage = `Error deleting this affiliation.`
+        const deletedAffiliationId = get(res, 'payload.data.deleteAffiliation')
+        if (deletedAffiliationId) {
+          successMessage = `Your affiliation was deleted.`
+          affiliations.items = affiliations.items.filter((a) => a.id !== deletedAffiliationId)
+        }
+        return this.setState({ affiliations, errorMessage, successMessage })
+      })
   }
 
-  const { community } = membership
+  leaveCommunity = (community) => {
+    const { leaveCommunity } = this.props
+    let { memberships } = this.state
+
+    leaveCommunity(community.id)
+      .then(res => {
+        let errorMessage, successMessage
+        if (res.error) errorMessage = `Error leaving ${community.name || 'this community'}.`
+        const deletedCommunityId = get(res, 'payload.data.leaveCommunity')
+        if (deletedCommunityId) {
+          successMessage = `You left ${community.name || 'this community'}.`
+          memberships = memberships.filter((m) => m.community.id !== deletedCommunityId)
+        }
+        return this.setState({ memberships, errorMessage, successMessage })
+      })
+  }
+
+  saveAffiliation = ({ role, preposition, orgName, url }) => {
+    const { affiliations } = this.state
+    this.props.createAffiliation({ role, preposition, orgName, url })
+      .then(res => {
+        let errorMessage, successMessage
+        if (res.error) errorMessage = get(res, 'payload.message', 'Error adding your affiliation.')
+        const affiliation = get(res, 'payload.data.createAffiliation')
+        if (affiliation) {
+          successMessage = `Your affiliation was added`
+          affiliations.items.push(affiliation)
+        }
+        return this.setState({ affiliations, errorMessage, successMessage, showAddAffiliations: !!errorMessage })
+      })
+  }
+
+  resetMessage = () => {
+    this.setState({ errorMessage: undefined, successMessage: undefined })
+  }
+
+  toggleAddAffiliations = () => {
+    this.setState({ showAddAffiliations: !this.state.showAddAffiliations })
+  }
+}
+
+export function AddAffiliation ({ close, save }) {
+  const PREPOSITIONS = [
+    { id: 'of', label: 'of' },
```

Comment:
why do we need this if the id and label are always the same? imagining a future where they might not be? could just be an array for now